runtime/jvmti/jvmtiObject.c

```diff
@@ -206,59 +209,72 @@ jvmtiGetObjectMonitorUsage(jvmtiEnv* env,
 	TRACE_FOUR_JVMTI_RETURN(jvmtiGetObjectMonitorUsage2, rv_owner, rv_entry_count, rv_notify_waiter_count, rv_waiter_count);
 }
 
-
 static void
-findMonitorThreads(J9VMThread * vmThread, J9JVMTIMonitorStats * pStats)
+findMonitorThreads(J9VMThread *vmThread, J9JVMTIMonitorStats *pStats)
 {
-	j9object_t lockObject;
+	j9object_t lockObject = NULL;
+	J9VMThread *currentThread = (J9VMThread *)pStats->currentThread;
```

Comment:
This needs to be conditional (Java 23+):
```
[2024-07-22T15:26:57.125Z] /Users/jenkins/workspace/Build_JDK21_aarch64_mac_Personal/openj9/runtime/jvmti/jvmtiObject.c:216:14: error: unused variable 'currentThread' [-Werror,-Wunused-variable]
[2024-07-22T15:26:57.125Z]         J9VMThread *currentThread = (J9VMThread *)pStats->currentThread;
[2024-07-22T15:26:57.125Z]                     ^
[2024-07-22T15:26:57.125Z] 1 error generated.
[2024-07-22T15:26:57.125Z] make[6]: *** [runtime/jvmti/CMakeFiles/j9jvmti.dir/jvmtiObject.c.o] Error 1
```

See https://openj9-jenkins.osuosl.org/job/Build_JDK21_aarch64_mac_Personal/154.

Reply:
Opened https://github.com/eclipse-openj9/openj9/pull/19896 to address the above issue.